app/views.py

```diff
@@ -121,3 +131,90 @@ def unauthorized():
     response.status_code = STATUS_CODE['UNAUTHORIZED']
     return response
 
+def validate_new_reservation(data):
+    response = validate_make_new_reservation_payload_format(data)
+    if response:
+        return response
+
+    startTime = int(data['startTime'])
+    endTime = int(data['endTime'])
+    response = validate_make_new_reservation_times(startTime, endTime)
+    if response:
+        return response
+
+    dateList = str(data['date']).split('-')
+    response = validate_make_new_reservation_date(dateList)
+    if response:
+        return response
+
+    roomId = data['roomId']
+    userId = data['userId']
+
+    response = validate_make_new_reservation_room_user_exists(roomId, userId)
+    if response:
+        return response
+
+    # TO DO: Actually save a reservation and return a useful confirmation message
+    return jsonify({'temp':'temp'})
+
+def validate_make_new_reservation_payload_format(data):
+    if 'roomId' not in data or \
+       'userId' not in data or \
+       'startTime' not in data or \
+       'endTime' not in data or \
+       'date' not in data or \
+       'description' not in data:
+        response = jsonify({'makeNewReservation error': '`roomId`, `userId`, `startTime`, `endTime`, `date` and `description` fields are required.'})
+        response.status_code = STATUS_CODE['UNPROCESSABLE']
+        return response
+
+    if not str(data['startTime']).isdigit() or \
+       not str(data['endTime']).isdigit() or \
+       not str(data['userId']).isdigit():
+        response = jsonify({'makeNewReservation error': '`userId`, `startTime` and `endTime` must be integers.'})
+        response.status_code = STATUS_CODE['UNPROCESSABLE']
+        return response
+
+def validate_make_new_reservation_times(startTime, endTime):
+    if startTime < 0 or startTime > 23 or endTime < 0 or endTime > 23:
+        response = jsonify({'makeNewReservation error': '`startTime` and `endTime` must be integers between 0 and 23.'})
+        response.status_code = STATUS_CODE['UNPROCESSABLE']
+        return response
+
+    if (startTime > endTime and endTime - startTime + 24 > 3) or \
```

Comment:
To deal with the special case of start time greater than end time, we need to use the modulus operation, because the time has wrapped around the 24-hour mark...

`if (end-start) % 24 > 3:
    return UNPROCESSABLE `